src/main/java/spoon/reflect/visitor/ImportScannerImpl.java

```diff
@@ -404,7 +362,7 @@ protected boolean isImportedInClassImports(CtTypeReference<?> ref) {
 	 */
 	private boolean declaringTypeIsLocalOrImported(CtTypeReference declaringType) {
 		if (declaringType != null) {
-			if (isImportedInClassImports(declaringType) || classNamePresentInJavaLang(declaringType)) {
+			if (!isTypeInCollision(declaringType, false) && (isImportedInClassImports(declaringType) || classNamePresentInJavaLang(declaringType))) {
```

Comment:
Actually I was testing that and trying to import the declaring type just below. So I assume you had a problem somewhere importing a declaring type: the condition below is useless then...